e2e/awspcaissuer_test.go

```diff
@@ -0,0 +1,237 @@
+		if regionExists {
+			testContext.region = region
+		}
+
+		cfg, cfgErr := config.LoadDefaultConfig(ctx, config.WithRegion(testContext.region))
+		if cfgErr != nil {
+			panic(cfgErr.Error())
+		}
+
+		testContext.iclient, err = clientV1beta1.NewForConfig(clientConfig)
+
+		if err != nil {
+			panic(err.Error())
+		}
+
+		testContext.cmClient, err = cmclientv1.NewForConfig(clientConfig)
+
+		if err != nil {
+			panic(err.Error())
+		}
+
+		// Create CAs to be used in testing
+		testContext.caArns["RSA"] = createCertificateAuthority(ctx, cfg, true)
+		log.Printf("Created RSA CA with arn %s", testContext.caArns["RSA"])
+
+		testContext.caArns["ECDSA"] = createCertificateAuthority(ctx, cfg, false)
+		log.Printf("Created EC CA with arn %s", testContext.caArns["ECDSA"])
+
+		xaRole, xaRoleExists := os.LookupEnv("PLUGIN_CROSS_ACCOUNT_ROLE")
+		if xaRoleExists {
+			testContext.xaCfg = assumeRole(ctx, cfg, xaRole, testContext.region)
+
+			testContext.caArns["XA"] = createCertificateAuthority(ctx, testContext.xaCfg, true)
+
+			log.Printf("Created XA CA with arn %s", testContext.caArns["XA"])
+
+			endEntityResourcePermission := "arn:aws:ram::aws:permission/AWSRAMDefaultPermissionCertificateAuthority"
+			subordinateCaResourcePermission := "arn:aws:ram::aws:permission/AWSRAMSubordinateCACertificatePathLen0IssuanceCertificateAuthority"
+
+			testContext.endEntityResourceShareArn = shareCA(ctx, cfg, testContext.xaCfg, testContext.caArns["XA"], endEntityResourcePermission)
+			testContext.subordinateCaResourceShareArn = shareCA(ctx, cfg, testContext.xaCfg, testContext.caArns["XA"], subordinateCaResourcePermission)
+		} else {
+			log.Print("Cross account role not present in PLUGIN_CROSS_ACCOUNT_ROLE, skipping cross account testing")
+		}
+
+		// Create an Access Key to be used for validiting auth via secret for an Issuer
+		userName, envUserExists := os.LookupEnv("PLUGIN_USER_NAME_OVERRIDE")
+
+		if !envUserExists {
+			testContext.userName, testContext.policyArn = createUser(ctx, cfg)
+			log.Printf("Created User %s with policy arn %s", testContext.userName, testContext.policyArn)
+		} else {
+			testContext.userName = userName
+			log.Printf("Using User %s from PLUGIN_USER_NAME_OVERRIDE", testContext.userName)
+		}
+
+		testContext.accessKey, testContext.secretKey = createAccessKey(ctx, cfg, testContext.userName)
+	})
+
+	// This is an AfterAll hook that will clean up the variables in the TestContext struct
+	suiteCtx.AfterSuite(func() {
+		ctx := context.TODO()
+
+		cfg, cfgErr := config.LoadDefaultConfig(ctx, config.WithRegion(testContext.region))
+
+		if cfgErr != nil {
+			panic(cfgErr.Error())
+		}
+
+		deleteCertificateAuthority(ctx, cfg, testContext.caArns["RSA"])
+		log.Printf("Deleted the RSA CA")
+
+		deleteCertificateAuthority(ctx, cfg, testContext.caArns["ECDSA"])
+		log.Printf("Deleted the EC CA")
+
+		//Delete IAM User and policy
+		deleteAccessKey(ctx, cfg, testContext.userName, testContext.accessKey)
```

Comment:
I'd like us to start tracking via UserAgent how many people are actually using Secret/Access Key auth. If low, should we deprecate it? It's not best practice at all to have long-lived credentials sitting around.